src/PolyhedralGeometry/iterators.jl

```diff
@@ -215,14 +215,22 @@ for (sym, name) in (("point_matrix", "Point Matrix"), ("vector_matrix", "Vector
     end
 end
 
-function matrix_for_polymake(iter::SubObjectIterator)
+function matrix_for_polymake(iter::SubObjectIterator; homogenized=false)
     if hasmethod(_matrix_for_polymake, Tuple{Val{iter.Acc}})
-        return _matrix_for_polymake(Val(iter.Acc))(Val(iter.Acc), iter.Obj; iter.options...)
+        return _matrix_for_polymake(Val(iter.Acc))(Val(iter.Acc), iter.Obj; homogenized=homogenized, iter.options...)
     else
         throw(ArgumentError("Matrix for Polymake not defined in this context."))
     end
 end
 
+# primitive generators only for ray based iterators
+matrix(::FlintIntegerRing, iter::SubObjectIterator{RayVector{Polymake.Rational}}) =
+    matrix(ZZ, Polymake.common.primitive(matrix_for_polymake(iter)))
+matrix(::FlintIntegerRing, iter::SubObjectIterator{<:Union{RayVector{Polymake.Integer},PointVector{Polymake.Integer}}}) =
+    matrix(ZZ, matrix_for_polymake(iter))
+matrix(::FlintRationalField, iter::SubObjectIterator{<:Union{RayVector,PointVector}}) =
+    matrix(QQ, Matrix{fmpq}(matrix_for_polymake(iter)))
```

Comment:
Unfortunately `FlintIntegerRing` and `FlintRationalField` are not true singletons (see https://github.com/Nemocas/Nemo.jl/pull/1221 and https://github.com/Nemocas/Nemo.jl/pull/1246). So formally, this is what one ought to do:

```suggestion
matrix(R::FlintIntegerRing, iter::SubObjectIterator{RayVector{Polymake.Rational}}) =
    matrix(R, Polymake.common.primitive(matrix_for_polymake(iter)))
matrix(R::FlintIntegerRing, iter::SubObjectIterator{<:Union{RayVector{Polymake.Integer},PointVector{Polymake.Integer}}}) =
    matrix(R, matrix_for_polymake(iter))
matrix(R::FlintRationalField, iter::SubObjectIterator{<:Union{RayVector,PointVector}}) =
    matrix(R, Matrix{fmpq}(matrix_for_polymake(iter)))
```

Reply:
Thanks, I wasn't aware of these details.